Document: torch.compile support limitations for Python

bilibili12433014 opened the pull request:
Added warning about Python compatibility with torch.compile in document.

bilibili12433014 committed:
Document torch.compile support limitations for Python 3.14

Added warning about Python 3.14 compatibility with torch.compile.

zou3519 commented on docs/source/torch.compiler.md:
```diff
@@ -34,6 +34,23 @@ In some cases, the terms `torch.compile`, TorchDynamo, `torch.compiler`
+`torch.compile` is **not supported on Python 3.14+**.
+
+If you attempt to use `@torch.compile` on Python 3.14 or newer,
+you will encounter an error similar to:
```
@williamwen42 @guilhermeleobas is this coming soon?

bilibili12433014 committed:
Update Python version support for torch.compile

bilibili12433014 committed:
Merge remote-tracking branch 'upstream/main' into patch-1

williamwen42 reviewed (commented):
PyTorch as a whole isn't supported on 3.15 - unsure if we need to specifically single out torch.compile.

bilibili12433014 committed:
Update warning for torch.compile Python version support

Clarify the warning regarding Python version support for torch.compile and remove the reference to a specific version number

williamwen42 reviewed (approved):
minor nit but otherwise lgtm

bilibili12433014 committed:
Update docs/source/user_guide/torch_compiler/torch.compiler.md

Co-authored-by: William Wen <william.wen42@gmail.com>

bilibili12433014 commented:
@pytorchmergebot merge

bilibili12433014 commented:
@pytorchmergebot merge -f "Minor update to fix doc."

bilibili12433014 commented:
Can you help me merge this pr? I do NOT have permissions to run workflows and "-f" is NOT available for me. If possible, can you directly run bot with "-f" as this is a minor update to fix doc.
@williamwen42 @zou3519 @guilhermeleobas
